File: Data Structures & Algorithms/two-integer-sum-ii/submission-4.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        n = len(numbers)
        p1 = 0
        p2 = n-1

        while p1 <= p2:
            sum = numbers[p1] + numbers[p2]
            if sum == target:
                return [p1+1, p2+1]
            elif sum < target:
                p1 += 1
            else:
                p2 -= 1
        

File: Data Structures & Algorithms/two-integer-sum/submission-4.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> mapping = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (mapping.containsKey(target - nums[i])) {
                return new int[]{mapping.get(target - nums[i]), i};
            }
            mapping.put(nums[i],i);
        }
        return new int[]{};
    }
}
